src/mainwindow.cpp

```diff
@@ -8948,9 +8948,15 @@ void MainWindow::handleScriptingNotesTagUpdating() {
 
     // workaround when signal blocking doesn't work correctly
     directoryWatcherWorkaround(true, true);
-
-    const QVector<Note> &notes = Note::fetchAll();
+    const QString defExt = Note::defaultNoteFileExtension();
+    const QStringList exts = Note::customNoteFileExtensionList(QStringLiteral("."));
+    const QVector<Note> notes = Note::fetchAll();
     for (const Note &note : notes) {
+        // ignore non-note files
+        const auto suffix = note.fileNameSuffix();
+        if (defExt != suffix || !exts.contains(suffix)) {
```

Comment:
> No idea about whether its included or not.

The answer is "yes", it's included. I just checked.

But what are you trying to achieve with this check? Why should there be non-note files in `Note::fetchAll()`?

Reply:
See my previous comment. For me the custom one is empty.

Reply:
> But what are you trying to achieve with this check? Why should there be non-note files in Note::fetchAll()?

Good question. Don't know the answer though, but I guess the code can tell.

Reply:
> I guess the code can tell

what code? did you find none-notes in the note list? what files?

Reply:
JPEG files

Reply:
referenced by a note.

Reply:
Whaaaat? 😆

Reply:
maybe because your Note::customNoteFileExtensionList is empty, can you please try to add some extension in the general settings.

Reply:
in addition I'll add a fix for you in customNoteFileExtensionList:

```cpp
    if (list.isEmpty()) {
        list << defaultNoteFileExtension();
    }
```

Reply:
> referenced by a note.

but why was it in the note list?

Reply:
Can you please check if the latest development branch still has JPEG files in the note list?

Reply:
just a note. not note list.

can check later.

Reply:
Did you? 😁

Reply:
can't reproduce anymore. I guess there wasn't a bug or some other misconfig.

Reply:
Great